Build failure: vmware-host-modules

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

1. Try to update kernel to 6.15

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/9w7qv9q06r00s0hwv4ffzqz8zv0yjlf4-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
substituteStream() in derivation vmware-modules-workstation-17.6.3-6.15: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'Makefile')
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j16 SHELL=/nix/store/xy4jjgw87sbgwylm5kn047d9gkbhsr9x-bash-5.2p37/bin/bash
make -C vmmon-only 
make -C vmnet-only 
make[1]: Entering directory '/build/source/vmnet-only'
make[1]: Entering directory '/build/source/vmmon-only'
Using kernel build system.
Using kernel build system.
make -C /nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/build/include/.. M=$PWD SRCROOT=$PWD/. \
  MODULEBUILDDIR= modules
make[2]: Entering directory '/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/build'
make[3]: Entering directory '/build/source/vmmon-only'
make -C /nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/build/include/.. M=$PWD SRCROOT=$PWD/. \
  MODULEBUILDDIR= modules
make[2]: Entering directory '/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/build'
make[3]: Entering directory '/build/source/vmnet-only'
  CC [M]  linux/driver.o
  CC [M]  linux/driverLog.o
  CC [M]  linux/hostif.o
  CC [M]  common/apic.o
  CC [M]  common/comport.o
  CC [M]  common/cpuid.o
  CC [M]  common/crosspage.o
  CC [M]  common/memtrack.o
  CC [M]  common/moduleloop.o
  CC [M]  common/phystrack.o
  CC [M]  common/sharedAreaVmmon.o
  CC [M]  common/statVarsVmmon.o
  CC [M]  common/task.o
[01m[Klinux/driver.c:21:10:[m[K [01;31m[Kfatal error: [m[Kdriver-config.h: No such file or directory
   21 | #include [01;31m[K"driver-config.h"[m[K
      |          [01;31m[K^~~~~~~~~~~~~~~~~[m[K
compilation terminated.
[01m[Klinux/hostif.c:30:10:[m[K [01;31m[Kfatal error: [m[Kdriver-config.h: No such file or directory
   30 | #include [01;31m[K"driver-config.h"[m[K
      |          [01;31m[K^~~~~~~~~~~~~~~~~[m[K
compilation terminated.
  CC [M]  common/vmx86.o
[01m[Klinux/driverLog.c:26:10:[m[K [01;31m[Kfatal error: [m[Kdriver-config.h: No such file or directory
   26 | #include [01;31m[K"driver-config.h"[m[K
      |          [01;31m[K^~~~~~~~~~~~~~~~~[m[K
compilation terminated.
  CC [M]  bootstrap/bootstrap.o
[01m[Kcommon/apic.c:20:10:[m[K [01;31m[Kfatal error: [m[Kvm_basic_defs.h: No such file or directory
   20 | #include [01;31m[K"vm_basic_defs.h"[m[K
      |          [01;31m[K^~~~~~~~~~~~~~~~~[m[K
compilation terminated.
make[5]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: linux/driver.o] Error 1
make[5]: *** Waiting for unfinished jobs....
make[5]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: common/apic.o] Error 1
make[5]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: linux/driverLog.o] Error 1
make[5]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: linux/hostif.o] Error 1
  CC [M]  driver.o
  CC [M]  hub.o
In file included from [01m[Kcommon/comport.c:40[m[K:
[01m[Kcommon/comport.h:25:10:[m[K [01;31m[Kfatal error: [m[KincludeCheck.h: No such file or directory
   25 | #include [01;31m[K"includeCheck.h"[m[K
      |          [01;31m[K^~~~~~~~~~~~~~~~[m[K
compilation terminated.
make[5]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: common/comport.o] Error 1
[01m[Kcommon/cpuid.c:22:13:[m[K [01;31m[Kfatal error: [m[Kdriver-config.h: No such file or directory
   22 | #   include [01;31m[K"driver-config.h"[m[K
      |             [01;31m[K^~~~~~~~~~~~~~~~~[m[K
compilation terminated.
  CC [M]  userif.o
  CC [M]  netif.o
make[5]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: common/cpuid.o] Error 1
[01m[Kcommon/crosspage.c:50:13:[m[K [01;31m[Kfatal error: [m[Kdriver-config.h: No such file or directory
   50 | #   include [01;31m[K"driver-config.h"[m[K
      |             [01;31m[K^~~~~~~~~~~~~~~~~[m[K
compilation terminated.
[01m[Kcommon/moduleloop.c:30:13:[m[K [01;31m[Kfatal error: [m[Kdriver-config.h: No such file or directory
   30 | #   include [01;31m[K"driver-config.h"[m[K
      |             [01;31m[K^~~~~~~~~~~~~~~~~[m[K
compilation terminated.
[01m[Kcommon/phystrack.c:38:13:[m[K [01;31m[Kfatal error: [m[Kdriver-config.h: No such file or directory
   38 | #   include [01;31m[K"driver-config.h"[m[K
      |             [01;31m[K^~~~~~~~~~~~~~~~~[m[K
compilation terminated.
  CC [M]  bridge.o
make[5]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: common/crosspage.o] Error 1
make[5]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: common/phystrack.o] Error 1
make[5]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: common/moduleloop.o] Error 1
  CC [M]  procfs.o
[01m[Kcommon/memtrack.c:79:13:[m[K [01;31m[Kfatal error: [m[Kdriver-config.h: No such file or directory
   79 | #   include [01;31m[K"driver-config.h"[m[K
      |             [01;31m[K^~~~~~~~~~~~~~~~~[m[K
compilation terminated.
[01m[Kbootstrap/bootstrap.c:27:10:[m[K [01;31m[Kfatal error: [m[Kvm_basic_types.h: No such file or directory
   27 | #include [01;31m[K"vm_basic_types.h"[m[K
      |          [01;31m[K^~~~~~~~~~~~~~~~~~[m[K
compilation terminated.
make[5]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: common/memtrack.o] Error 1
  CC [M]  smac_compat.o
[01m[Kcommon/sharedAreaVmmon.c:27:13:[m[K [01;31m[Kfatal error: [m[Kdriver-config.h: No such file or directory
   27 | #   include [01;31m[K"driver-config.h"[m[K
      |             [01;31m[K^~~~~~~~~~~~~~~~~[m[K
compilation terminated.
make[5]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: bootstrap/bootstrap.o] Error 1
  CC [M]  smac.o
[01m[Kcommon/statVarsVmmon.c:27:13:[m[K [01;31m[Kfatal error: [m[Kdriver-config.h: No such file or directory
   27 | #   include [01;31m[K"driver-config.h"[m[K
      |             [01;31m[K^~~~~~~~~~~~~~~~~[m[K
compilation terminated.
[01m[Kcommon/task.c:38:13:[m[K [01;31m[Kfatal error: [m[Kdriver-config.h: No such file or directory
   38 | #   include [01;31m[K"driver-config.h"[m[K
      |             [01;31m[K^~~~~~~~~~~~~~~~~[m[K
compilation terminated.
make[5]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: common/sharedAreaVmmon.o] Error 1
make[5]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: common/task.o] Error 1
make[5]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: common/statVarsVmmon.o] Error 1
  CC [M]  vnetEvent.o
[01m[Kcommon/vmx86.c:29:13:[m[K [01;31m[Kfatal error: [m[Kdriver-config.h: No such file or directory
   29 | #   include [01;31m[K"driver-config.h"[m[K
      |             [01;31m[K^~~~~~~~~~~~~~~~~[m[K
compilation terminated.
  CC [M]  vnetUserListener.o
make[5]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:203: common/vmx86.o] Error 1
make[4]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/Makefile:2003: .] Error 2
make[3]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/Makefile:248: __sub-make] Error 2
make[3]: Leaving directory '/build/source/vmmon-only'
make[2]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/Makefile:248: __sub-make] Error 2
make[2]: Leaving directory '/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/build'
make[1]: *** [Makefile:117: vmmon.ko] Error 2
make[1]: Leaving directory '/build/source/vmmon-only'
make: *** [Makefile:21: vmmon-only] Error 2
make: *** Waiting for unfinished jobs....
  LD [M]  vmnet.o
vmnet.o: warning: objtool: VNetCsumAndCopyToUser+0x4d: call to csum_partial_copy_nocheck() with UACCESS enabled
vmnet.o: error: objtool: init_module(): Magic init_module() function name is deprecated, use module_init(fn) instead
make[5]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/scripts/Makefile.build:408: vmnet.o] Error 255
make[5]: *** Deleting file 'vmnet.o'
make[4]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/Makefile:2003: .] Error 2
make[3]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/Makefile:248: __sub-make] Error 2
make[3]: Leaving directory '/build/source/vmnet-only'
make[2]: *** [/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/source/Makefile:248: __sub-make] Error 2
make[2]: Leaving directory '/nix/store/gl4zb4minm4ddaz31lhsx2rr964z1gjl-linux-6.15-dev/lib/modules/6.15.0/build'
make[1]: *** [Makefile:117: vmnet.ko] Error 2
make[1]: Leaving directory '/build/source/vmnet-only'
make: *** [Makefile:21: vmnet-only] Error 2
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.14.8, NixOS, 25.11 (Xantusia), 25.11.20250525.62b852f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.1
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/karun/.config/nix/nix.conf:/nix/store/bvzbm6dd9di2kh0lqpsw1169cvm3vcyd-stylix-kde-config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/karun/.local/share/flatpak/exports/etc/xdg/nix/nix.conf:/var/lib/flatpak/exports/etc/xdg/nix/nix.conf:/home/karun/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/karun/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/karun/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/05pfcid57kviaqpr4i57ydwmdvgdigxz-lix-2.91.1/share`
 - nixpkgs: `/nix/store/nfqw3h3qc2mq37v6amw80b1rpi0g9x61-source`

### Notify maintainers

@deinferno @vifino

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/vmware/default.nix
+++ b/pkgs/os-specific/linux/vmware/default.nix
@@ -41,13 +41,13 @@ stdenv.mkDerivation {
     mkdir -p "$out/lib/modules/${kernel.modDirVersion}/misc"
   '';
 
-  meta = with lib; {
+  meta = {
     description = "Kernel modules needed for VMware hypervisor";
     homepage = "https://github.com/mkubecek/vmware-host-modules";
-    license = licenses.gpl2Only;
+    license = lib.licenses.gpl2Only;
     platforms = [ "x86_64-linux" ];
     broken = (kernel.kernelOlder "5.5" && kernel.isHardened);
-    maintainers = with maintainers; [
+    maintainers = with lib.maintainers; [
       deinferno
       vifino
     ];

--- a/pkgs/os-specific/linux/vmware/default.nix
+++ b/pkgs/os-specific/linux/vmware/default.nix
@@ -9,13 +9,13 @@
 
 stdenv.mkDerivation {
   pname = "vmware-modules";
-  version = "workstation-17.6.3-${kernel.version}";
+  version = "workstation-17.6.3-20250608-${kernel.version}";
 
   src = fetchFromGitHub {
     owner = "philipl";
     repo = "vmware-host-modules";
-    tag = "w17.6.3";
-    hash = "sha256-n9aLpHcO7m51eRtcWWBfTpze0JIWvne+UcYACoA5afc=";
+    rev = "e08764f84df3bd14407291e02346b7c5c54d0bf6";
+    hash = "sha256-RDVKropC+VOv1uJijOGlLt6iqZI8Cm9aWlbmvZj2CSk=";
   };
 
   hardeningDisable = [ "pic" ];

